### PR TITLE
[orchagent]: Support LAG_MEMBER_TABLE

### PR DESCRIPTION
**Pull request by stcheng**
- Add LAG_MEMBER_TABLE in teamsyncd
- Add LAG_MEMBER_TABLE in orchagent

Signed-off-by: Shuotian Cheng <shuche@microsoft.com>

### CODE REVIEW

**Comment by stcheng** [comment]
To address issue: https://github.com/Azure/sonic-swss/issues/160

**Comment by stcheng** [comment]
```
127.0.0.1:6379> keys LAG*                               
 1) "LAG_MEMBER_TABLE:PortChannel56:Ethernet56"         
 2) "LAG_TABLE:PortChannel40"                           
 3) "LAG_MEMBER_TABLE:PortChannel0:Ethernet4"           
 4) "LAG_MEMBER_TABLE:PortChannel32:Ethernet36"         
 5) "LAG_MEMBER_TABLE:PortChannel8:Ethernet8"           
 6) "LAG_TABLE:PortChannel32"                           
 7) "LAG_TABLE:PortChannel16"                           
 8) "LAG_MEMBER_TABLE:PortChannel8:Ethernet12"          
 9) "LAG_MEMBER_TABLE:PortChannel48:Ethernet52"         
10) "LAG_MEMBER_TABLE:PortChannel24:Ethernet28"         
11) "LAG_MEMBER_TABLE:PortChannel16:Ethernet16"         
12) "LAG_TABLE:PortChannel0"                            
13) "LAG_MEMBER_TABLE:PortChannel0:Ethernet0"           
14) "LAG_MEMBER_TABLE:PortChannel40:Ethernet40"         
15) "LAG_MEMBER_TABLE:PortChannel48:Ethernet48"         
16) "LAG_MEMBER_TABLE:PortChannel40:Ethernet44"         
17) "LAG_MEMBER_TABLE:PortChannel32:Ethernet32"         
18) "LAG_TABLE:PortChannel48"                           
19) "LAG_TABLE:PortChannel24"                           
20) "LAG_MEMBER_TABLE:PortChannel56:Ethernet60"         
21) "LAG_MEMBER_TABLE:PortChannel24:Ethernet24"         
22) "LAG_TABLE:PortChannel56"                           
23) "LAG_MEMBER_TABLE:PortChannel16:Ethernet20"         
24) "LAG_TABLE:PortChannel8"                            ```

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -529,142 +529,142 @@ void PortsOrch::doLagTask(Consumer &consumer)
...
+
+void PortsOrch::doLagMemberTask(Consumer &consumer)
+{
+    if (!isInitDone())
+        return;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >KeyOpFieldsValuesTuple [](start = 8, length = 22)

use reference? no copy

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -529,142 +529,142 @@ void PortsOrch::doLagTask(Consumer &consumer)
...
+
+        /* Retrieve LAG alias and LAG member alias from key */
+        string key = kfvKey(t);
+        size_t found = key.find(':');
+        string lag_alias = key.substr(0, found);
+        string port_alias = key.substr(found+1);
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >found [](start = 39, length = 5)

handle invalid index before indexing?

**Comment by qiluo-msft** [comment]
>     if (table_name == APP_PORT_TABLE_NAME)

A minor optimization opportunity: use a map<string, func> to simulate switch-case behavior. Implement the map at initialization.

---
Refers to: orchagent/portsorch.cpp:678 in 6a99612. [](commit_id = 6a996122348875d61aba46470810cfe96baa7fe6, deletion_comment = False)

File: teamsyncd/teamsync.cpp [hunk 3]
```diff
@@ -65,7 +66,7 @@ void TeamSync::addLag(const string &lagName, int ifindex, bool admin_state,
         return;
 
     /* Start track the team instance */
-    TeamPortSync *sync = new TeamPortSync(lagName, ifindex, &m_lagTable);
+    TeamPortSync *sync = new TeamPortSync(lagName, ifindex, &m_lagMemberTable);
     m_select->addSelectable(sync);
     m_teamPorts[lagName] = shared_ptr<TeamPortSync>(sync);
```
Comment (qiluo-msft) [inline teamsyncd/teamsync.cpp]: >shared_ptr [](start = 27, length = 10)

If the raw pointer is going to be inside a shared_ptr, why not use make_shared() at the beginning, and make it more exception-safe.

**Comment by lguohan** [comment]
:shipit:

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (stcheng) [inline orchagent/portsorch.cpp]: this is a very light-weighted copy. i just follow the rest of the code.

File: teamsyncd/teamsync.cpp (hunk 3 above)
Comment (stcheng) [inline teamsyncd/teamsync.cpp]: i think the reason is that the original author of this code wanted to pass "Selectable *" to addSelectable() function as this function right now only supports weak_ptr.
anyway, this is a separate issue from this pull request.

**Comment by stcheng** [comment]
@qiluo-msft the comments you left are not related to the purpose of this pull request. i will add different pull requests later to address them.

**Comment by lguohan** [comment]
@qiluo-msft, please help to close this.

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: Not that light. Suggestion "auto& t".

---
In reply to: [111024651](https://github.com/Azure/sonic-swss/pull/187#discussion_r111024651) [](ancestors = 111024651)

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (stcheng) [inline orchagent/portsorch.cpp]: fixed

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (stcheng) [inline orchagent/portsorch.cpp]: fixed.

File: teamsyncd/teamsync.cpp (hunk 3 above)
Comment (stcheng) [inline teamsyncd/teamsync.cpp]: fixed.

**Pull request merged**